src/main/java/seedu/address/logic/commands/UnlinkCommand.java

```diff
@@ -162,7 +162,9 @@ public boolean equals(Object other) {
         }
 
         UnlinkCommand otherCommand = (UnlinkCommand) other;
-        return personIndex.equals(otherCommand.personIndex)
-                && internshipIndex.equals(otherCommand.internshipIndex);
+        return ((personIndex == null && otherCommand.personIndex == null)
+                || (personIndex != null && personIndex.equals(otherCommand.personIndex)))
```

Comment:
Perhaps `Objects.equals(personIndex, otherCommand.personIndex) && Objects.equals(internshipIndex, otherCommand.internshipIndex)` would be more succinct here?

Reply:
`Objects#equals()` handles the null cases for you